### PR TITLE
NT-0026 - Added user delete route + auth middleware

### PR DESCRIPTION
**Pull request by kamilmoskal**

### CODE REVIEW

File: middleware/auth.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  // Verify token
+  jwt
+    .verify(token, config.get("jwtSecret"))
+    .then(decoded => {
+      req.user = decoded.user;
+      next();
+    })
+    .catch(err => res.status(401).json({ msg: "Token is not valid" }));
```
Comment (mras2303) [inline middleware/auth.js]: I guess, it should be ```message.TOKEN_IS_INVALID```

File: middleware/auth.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const jwt = require("jsonwebtoken");
+const config = require("config");
+
+module.exports = (req, res, next) => {
+  // Get token from header
+  const token = req.header("x-auth-token");
+
+  // Check if not token
+  if (!token) {
+    return res.status(401).json({ msg: "No token, authorization denied" });
```
Comment (mras2303) [inline middleware/auth.js]: I guess, it should be ```message.NO_TOKEN_AUTHORIZATION_DENIED```

File: middleware/auth.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      .status(401)
+      .json({ msg: messages.NO_TOKEN_AUTHORIZATION_DENIED });
+  }
+
+  // Verify token
+  jwt.verify(token, config.get("jwtSecret"), (err, decoded) => {
+    if (err) {
+      res.status(401).json({ msg: messages.TOKEN_IS_INVALID });
+    } else {
+      req.user = decoded.user;
```
Comment (damian0o) [inline middleware/auth.js]: we can also verify that user confirm his email

File: controllers/users/users.js [hunk 4]
```diff
@@ -60,3 +60,9 @@ exports.registerUser = (req, res) => {
     }
   });
 };
+
+exports.deleteUser = (req, res) => {
+  User.findOneAndRemove({ _id: req.user.id })
```
Comment (damian0o) [inline controllers/users/users.js]: we should remove user from db or mark as deleted with some field?

File: middleware/auth.js (hunk 3 above)
Comment (kamilmoskal) [reply middleware/auth.js]: good idea

File: controllers/users/users.js (hunk 4 above)
Comment (kamilmoskal) [reply controllers/users/users.js]: we can discuss it and change later

**Pull request merged**